sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonParser.scala

```diff
@@ -298,22 +312,22 @@ class JacksonParser(
         // Here, we pass empty `PartialFunction` so that this case can be
         // handled as a failed conversion. It will throw an exception as
         // long as the value is not null.
-        parseJsonToken(parser, dataType)(PartialFunction.empty[JsonToken, Any])
+        parseJsonToken[AnyRef](parser, dataType)(PartialFunction.empty[JsonToken, AnyRef])
   }
 
   /**
    * This method skips `FIELD_NAME`s at the beginning, and handles nulls ahead before trying
    * to parse the JSON token using given function `f`. If the `f` failed to parse and convert the
    * token, call `failedConversion` to handle the token.
    */
-  private def parseJsonToken(
+  private def parseJsonToken[R >: Null](
```

Comment:
Yes, I said +1 because it explicitly expresses it should be nullable and I _assumed_ (because I did not check the byte codes by myself and I might be wrong) that it gives a hint to compiler because `Null` is `null`able (I remember I googled and played some references for whole several days before when I was investigating another null-related PR).